router/batchrouter/batchrouter.go

```diff
@@ -2023,14 +2009,23 @@ func (brt *HandleT) readAndProcess() {
 			brtQueryStat.End()
 			brt.logger.Debugf("BRT: %s: Length of jobs received: %d", brt.destType, len(jobs))
 
+			jobsByDesID := lo.GroupBy(jobs, func(job *jobsdb.JobT) string {
+				var parameters JobParametersT
+				if json.Unmarshal(job.Parameters, &parameters) != nil {
+					brt.logger.Error("BRT: %s: Unmarshal of job parameters failed. ", brt.destType, string(job.Parameters))
+				}
+				return parameters.DestinationID
+			})
 			var wg sync.WaitGroup
-			for destID, batchDest := range destinationsMap {
-				brt.setDestInProgress(destID, true)
-
-				wg.Add(1)
-				brt.processQ <- &BatchDestinationDataT{batchDestination: *batchDest, jobs: jobs, parentWG: &wg}
+			for destID, destJobs := range jobsByDesID {
+				if batchDest, ok := destinationsMap[destID]; ok {
+					brt.setDestInProgress(destID, true)
+					wg.Add(1)
+					brt.processQ <- &BatchDestinationDataT{batchDestination: *batchDest, jobs: destJobs, parentWG: &wg}
+				} else {
+					brt.logger.Errorf("BRT: %s: Destination %s not found in destinationsMap", brt.destType, destID)
```

Comment:
Although not relevant to this pr, we should do something about jobs of destination not found in destinationsMap. Abort?